src/Nncase.Tests/Core/UnitTestTensor.cs

```diff
@@ -1,54 +1,143 @@
 ﻿// Copyright (c) Canaan Inc. All rights reserved.
 // Licensed under the Apache license. See LICENSE file in the project root for full license information.
 
+using System;
+using System.Collections;
+using System.Collections.Immutable;
 using Nncase;
+using Nncase.IR;
 using Xunit;
 
 namespace Nncase.Tests.CoreTest;
 
 public sealed class UnitTestTensor
 {
     [Fact]
-    public void TestIsContiguousSlice()
+    public void TestICollection()
     {
-        var dim1 = new[] { 1, 512, 14, 14 };
+        var a = new float[] { 1, 2, 3, 4, 5, 6, 7, 8 };
+        var t = (ICollection)Tensor.From(a, new int[] { 1, 1, 2, 4 });
+        Assert.Equal(a.Length, t.Count);
+        Assert.False(t.IsSynchronized);
+        Assert.Equal((object)t, t.SyncRoot);
+    }
 
-        Assert.True(TensorUtilities.IsContiguousSlice(
-          dim1,
-          new[] { 0..1, 0..512, 0..14, 0..14 }));
+    [Fact]
+    public void TestIList()
+    {
+        var a = new float[] { 1, 2, 3, 4, 5, 6, 7, 8 };
+        var t = Tensor.From(a, new int[] { 1, 1, 2, 4 });
+        var list = (IList)t;
+        Assert.True(list.IsFixedSize);
+        Assert.False(list.IsReadOnly);
 
-        Assert.True(TensorUtilities.IsContiguousSlice(
-          dim1,
-          new[] { 0..1, 0..1, 0..1, 0..14 }));
+        Assert.Equal(1f, list[0]);
+        list[0] = 100f;
+        Assert.Equal(100f, list[0]);
 
-        Assert.True(TensorUtilities.IsContiguousSlice(
-          dim1,
-          new[] { 0..1, 0..1, 0..1, 7..14 }));
+        list.Clear();
+        var expected = Tensor.Zeros<float>(new int[] { 1, 1, 2, 4 });
+        Assert.Equal(expected.ToArray<float>(), t.ToArray<float>());
+    }
 
-        Assert.True(TensorUtilities.IsContiguousSlice(
-          dim1,
-          new[] { 0..1, 0..1, 7..14, 0..14 }));
+    [Fact]
+    public void TestIndices()
+    {
+        var a = new float[] { 1, 2, 3, 4, 5, 6, 7, 8 };
+        var t = Tensor.From(a, new int[] { 1, 1, 2, 4 });
+
+        Assert.Equal(7, t[new int[] { 0, 0, 1, 2 }]);
+        t[new int[] { 0, 0, 1, 2 }] = 700;
+        Assert.Equal(700, t[new int[] { 0, 0, 1, 2 }]);
+    }
 
-        Assert.False(TensorUtilities.IsContiguousSlice(
-          dim1,
-          new[] { 0..1, 0..512, 0..7, 0..14 }));
+    [Fact]
+    public void TestFromBytes1()
```

Comment:
a good name is useful for understand what is the target being tested, It is difficult to understand what tests have been done when the contents and names of the test bodies are very similar.

Reply:
These are 3 overloads of FromBytes(). Now merge them into one test case and add more comment